build/xtask/src/dist.rs

```diff
@@ -2078,24 +2086,24 @@ fn binary_to_srec(
         panic!("SREC limit of 2^24 output sections exceeded");
     }
 
-    srec_out.push(srec::Record::S7(srec::Address32(entry)));
+    srec_out.push(srec::Record::S7(srec::Address32(entry as u32)));
 
     let srec_image = srec::writer::generate_srec_file(&srec_out);
     std::fs::write(out, srec_image)?;
     Ok(())
 }
 
 fn write_srec(
-    sections: &BTreeMap<u32, LoadSegment>,
-    kentry: u32,
+    sections: &BTreeMap<AbiSize, LoadSegment>,
+    kentry: AbiSize,
     out: &Path,
 ) -> Result<()> {
     let mut srec_out = vec![srec::Record::S0("hubris".to_string())];
     for (&base, sec) in sections {
         // SREC record size limit is 255 (0xFF). 32-bit addressed records
         // additionally contain a four-byte address and one-byte checksum, for a
         // payload limit of 255 - 5.
-        let mut addr = base;
+        let mut addr = base as u32;
```

Comment:
Is there anything to check that the `base` will fit in a `u32`? If not, an error should be generated instead of silently ignoring that.

Reply:
Fixed in next patchset.